### ISSUE TITLE
Theme templates gets copied into deploy destdir

### ISSUE THREAD

**Issue by AdUser** [preaction/Statocles#399]
Tested on 0.055. Part of config with theme- and deploy-related settings:

```yaml
site:
  args:
    theme:
      $ref: theme
deploy:
  args:
    path: html
  class: Statocles::Deploy::File
theme:
  args:
    store: theme
  class: Statocles::Theme
```
Try to run deploy:

```
me@localhost:~/site$ statocles deploy
me@localhost:~/site$ find html/ -name '*.ep'
html/theme/blog/index.rss.ep
html/theme/blog/index.html.ep
html/theme/blog/index.atom.ep
html/theme/blog/post.html.ep
html/theme/site/sitemap.xml.ep
html/theme/site/robots.txt.ep
html/theme/site/layout.html.ep
```

It's not critical, but potentially may expose private data.

**Comment by preaction** [comment]
Yeah, that's mostly intentional: I didn't want to make a guess about which files in the `theme/` directory should be deployed or shouldn't be deployed, since you can put CSS, JavaScript, images, and other things in there that need to come along when deploying the site. And people would have to know where those templates are, since there are no links to them (counterpoint, though, if your deploy involves Github, your templates will be in a possibly-public Github repo for anyone to easily navigate to).

And different themes could have different things to filter. The default theme uses `/.ep$/` for templates, but another theme might use `/.tt$/` or something. So, to ensure nothing gets broken, I just left the templates in the deploy.

But, since they're code, and indeed in the `.ep` templates you can embed any Perl you want, I can agree there's the potential for very private information to be leaked. You could, for example, build a template that uses [DBI](http://metacpan.org/pod/DBI) to access a database (with username and password) to add data to the page. The page won't have the database's login information, but the template must.

And, now that I think about it, and I'm surprised I didn't think about it before: The templates _themselves_ are valuable. The site's design is yours. People should not be able to easily take that and put it in their own Statocles site without your permission, and that is what I've allowed by exposing all the templates to every website. There's nothing stopping them from taking the CSS, JS, and images and coming up with their own Statocles templates, but why make it easy for them if the author of the templates does not want to?

So, to address this, we should probably first make the `Statocles::Theme` compose the `Statocles::App` role, so that the `Theme` object has ultimate control over which things will be deployed to the live site (right now, this happens in the `Site` object, very messy...).

Once that's done, then we could have an `exclude` filter attribute that would allow for excluding pages that shouldn't be deployed. This could go on the `Theme` object, or on all `App` objects. It seems like a common enough thing to want: There could be all kinds of reasons why one might want to exclude a page from being deployed in any application.

**Comment by preaction** [comment]
This is now released in v0.056
